backend/src/org/commcare/session/CommCareSession.java

```diff
@@ -125,22 +126,21 @@ public CommCareSession(CommCareSession oldCommCareSession) {
      */
     private Vector<Entry> getEntriesForCommand(String commandId,
                                                OrderedHashtable<String, String> currentSessionData) {
+        Vector<Entry> entries = new Vector<>();
         for (Suite s : platform.getInstalledSuites()) {
-            for (Menu m : s.getMenus()) {
-                // We need to see if everything in this menu can be matched
-                if (commandId.equals(m.getId())) {
-                    return getEntriesFromMenu(m, currentSessionData);
+            List<Menu> menusWithId = s.getMenusWithId(commandId);
+            if (menusWithId != null) {
+                for (Menu menu : menusWithId) {
+                    entries.addAll(getEntriesFromMenu(menu, currentSessionData));
```

Comment:
Is this the fix? curious how this resolves Kriti's issue of displaying the wrong menu name. Or is that not what this PR is addressing? (tried to read through the ticket you linked but there were a few different things going on there)